expenses_management_tool/data_generator.py

```diff
@@ -0,0 +1,83 @@
+from faker import Faker
+import pandas as pd 
+import random
+
+
+fake = Faker()
+
+
+Category = ["Food and Beverages", "Consumables", "Leisure and Entertainment", "Transport", "Extraordinary Expenses"]
+Name = []
+Cost = []
+Date = []
+Importance = []
+GorB = ["Good", "Bad"]
+
+# Define a dictionary that maps categories to lists of subcategories
+subcategories = {
+    "Food and Beverages": ["Supermarkt", "Kiosk", "Bakery", "Market", "Other"],
+    "Consumables": ["Drugstore","Clothing", "Electronics", "Furniture", "Pets", "Education", "Stationery", "Medicines", "Narcotics", "Other"],
+    "Leisure and Entertainment": ["Cinema", "Restaurant", "Party", "Event", "Hairdresser", "Hobby", "Vacation", "Other"],
+    "Transport": ["Public Transportation", "Private Transportation", "Other"],
+    "Extraordinary Expenses": ["Additional Costs", "Investments", "Other"]
+}
+
+# Define a dictionary that maps categories to ranges of possible costs
+cost_ranges = {
+    "Food and Beverages": (5, 200),
+    "Consumables": (10, 450),
+    "Leisure and Entertainment": (20, 200),
+    "Transport": (5, 35),
+    "Extraordinary Expenses": (150, 700)
+}
+
+
+# Define the Subcategory based on the main Category
+def Subcategory(row):
+    # Look up the list of subcategories for the current category
+    subcat_list = subcategories.get(row["Category"], [])
+    
+    # Choose a random element from the list of subcategories
+    return random.choice(subcat_list)
+
+
+
+# Define a function that generates a random cost within the range for the given category
+def generate_cost(row):
+    # Look up the range of possible costs for the current category
+    cost_range = cost_ranges.get(row["Category"], (0, 0))
+    
+    # Generate a random cost within the range
+    cost = random.uniform(*cost_range)
+
+ # Round the cost to two decimal places
+    return round(cost, 2)
+
+
+
+def dummy_data(amount): 
+    data = []
+    for i in range(amount):
+        row = {}
+        row["Category"] = random.choice(Category)
+        row["Name"] = fake.name()

```

Comment:
"The Name values are randomly generated." 
->
to get the data generator up and running, that's perfectly fine. But we should try to give realistic data as a goal. But I can take care of that